installer/x86_64/install.sh

```diff
@@ -652,7 +652,7 @@ fi
 # Note: assume that apparmor is supported in the kernel
 demo_grub_entry="$demo_volume_revision_label"
 if [ "$install_env" = "sonic" ]; then
-    old_sonic_menuentry=$(cat /host/grub/grub.cfg | sed "/$running_sonic_revision/,/}/!d")
+    old_sonic_menuentry=$(cat /host/grub/grub.cfg | sed "/'${demo_volume_label}-${running_sonic_revision}'/,/}/!d")
```

Comment:
Also add the `^menuentry ` at the beginning of the pattern to be more strict.

Reply:
Fixed